lib/src/http/utils.dart

```diff
@@ -46,22 +48,77 @@ Duration calculateBackOff(RequestAutoRetry autoRetry) {
   return backOff;
 }
 
-Duration _calculateExponentialBackOff(RequestAutoRetry autoRetry) {
+Duration _calculateExponentialBackOff(RequestAutoRetry autoRetry,
+    {@visibleForTesting Random random}) {
+  if (autoRetry.backOff.withJitter ?? false) {
+    final jitteredBackOff = _calculateAdvancedExponentialJitteredBackOffInMs(
+        autoRetry,
+        random: random);
+    // If we're over the maximum duration, fall back to a fixed maxInterval with full jitter
+    if (jitteredBackOff > autoRetry.backOff.maxInterval.inMilliseconds) {
+      return Duration(
+          milliseconds:
+              (autoRetry.backOff.maxInterval.inMilliseconds.toDouble() *
+                      (random ?? Random()).nextDouble())
+                  .toInt());
+    }
+    return Duration(milliseconds: jitteredBackOff);
+  }
+
+  return Duration(
+      milliseconds: _calculateUnjitteredExponentialBackOffInMs(autoRetry));
+}
+
+int _calculateUnjitteredExponentialBackOffInMs(autoRetry) {
   int backOffInMs = autoRetry.backOff.interval.inMilliseconds *
       pow(_exponentialBase, autoRetry.numAttempts);
-  backOffInMs = min(autoRetry.backOff.maxInterval.inMilliseconds, backOffInMs);
+  return min(autoRetry.backOff.maxInterval.inMilliseconds, backOffInMs);
+}
 
-  if (autoRetry.backOff.withJitter == true) {
-    final random = Random();
-    backOffInMs = random.nextInt(backOffInMs);
+/// Returns the jittered backoff delay in ms using an advanced jittering algorithm.
+///
+/// Taken from https://github.com/Polly-Contrib/Polly.Contrib.WaitAndRetry/blob/master/src/Polly.Contrib.WaitAndRetry/Backoff.DecorrelatedJitterV2.cs#L35-L65
+/// See the details here: https://github.com/Polly-Contrib/Polly.Contrib.WaitAndRetry#wait-and-retry-with-jittered-back-off
+///
+/// See RFD 277 where this was originally proposed: https://sandbox.wdesk.com/a/QWNjb3VudB82NzE5NTMwMjcyOTQ4MjI0/doc/0c0c78a2bba448babb8bb2e45ba62f7b/r/-1/v/1/sec/0c0c78a2bba448babb8bb2e45ba62f7b_4
```

Comment:
Unsure about referencing this RFD here, since this is OSS 🤷‍♂

Reply:
Yeah, I'd vote for leaving it out